sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -213,7 +214,8 @@ object ColumnStat extends Logging {
         val nullLit = Literal(null, col.dataType)
         struct(
           ndv, nullLit, nullLit, numNulls,
-          Ceil(Average(Length(col))), Cast(Max(Length(col)), LongType))
+          Coalesce(Seq(Ceil(Average(Length(col))), defaultSize)),
```

Comment:
add a line of comment explaining we set the value to defaultSize if all the values are null (of if there is no value).